tests/vision/vision_conftest.py

```diff
@@ -129,6 +165,39 @@ def mnist_dataset_train_imgaug():
 def trained_yolov5_object_detection(device):  # pylint: disable=redefined-outer-name
     return load_yolov5_model(device=device)
 
+@pytest.fixture(scope='session')
+def mock_trained_mnist(device):  # pylint: disable=redefined-outer-name
+
+    class MockMnist:
+        def __call__(self, batch):
```

Comment:
instead of having a mock model and a real model, we should have combination that works like:
```
for img in batch:
  key = _hash_batch(batch)
  if key in mnist_predictions_dict:
     return mnist_predictions_dict[key]
  else:
     real_model.to(device)(img)

```